src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -13,6 +13,49 @@ public enum JsonCommentHandling : byte
         Disallow = (byte)0,
         Skip = (byte)2,
     }
+    public sealed partial class JsonDocument : System.IDisposable
```

Comment:
Is it going to be strange for folks to need to dispose an XxDocument?  Is that required for other XxDocument in the framework?  Is it required for any other JSON document-like types in other JSON libraries?

Reply:
Usually the DOM APIs _aren't_ IDisposable, but they have readers that are.  So "no, but nearly".  It seems like a natural consequence of the performance goals.

If it's "too strange for the surface area" then the next option that I see is to use generic List; the consequence of not disposing in the current model being that the document "stole" the array from the pool rather than rented it.